src/fps/proposal/MultisigProposal.sol

```diff
@@ -0,0 +1,854 @@
+    }
+
+    /// @notice helper function to prepare the signatures to be executed
+    /// @param _safe The address of the parent multisig
+    /// @param hash The hash to be approved
+    /// @return The signatures to be executed
+    function prepareSignatures(address _safe, bytes32 hash) internal view returns (bytes memory) {
+        // prepend the prevalidated signatures to the signatures
+        address[] memory approvers = getApprovers(_safe, hash);
+        return genPrevalidatedSignatures(approvers);
+    }
+
+    /// @notice helper function to generate the prevalidated signatures for a given list of addresses
+    function genPrevalidatedSignatures(address[] memory _addresses) internal pure returns (bytes memory) {
+        LibSort.sort(_addresses);
+        bytes memory signatures;
+        for (uint256 i; i < _addresses.length; i++) {
+            signatures = bytes.concat(signatures, genPrevalidatedSignature(_addresses[i]));
+        }
+        return signatures;
+    }
+
+    /// @notice helper function to generate the prevalidated signature for a given address
+    function genPrevalidatedSignature(address _address) internal pure returns (bytes memory) {
+        uint8 v = 1;
+        bytes32 s = bytes32(0);
+        bytes32 r = bytes32(uint256(uint160(_address)));
+        return abi.encodePacked(r, s, v);
+    }
+
+    /// @notice helper function to get the approvers for a given hash
+    function getApprovers(address _safe, bytes32 hash) internal view returns (address[] memory) {
+        // get a list of owners that have approved this transaction
+        IGnosisSafe safe = IGnosisSafe(_safe);
+        uint256 threshold = safe.getThreshold();
+        address[] memory owners = safe.getOwners();
+        address[] memory approvers = new address[](threshold);
+        uint256 approverIndex;
+        for (uint256 i; i < owners.length; i++) {
+            address owner = owners[i];
+            uint256 approved = safe.approvedHashes(owner, hash);
+            if (approved == 1) {
+                approvers[approverIndex] = owner;
+                approverIndex++;
+                if (approverIndex == threshold) {
+                    return approvers;
+                }
+            }
+        }
+        address[] memory subset = new address[](approverIndex);
+        for (uint256 i; i < approverIndex; i++) {
+            subset[i] = approvers[i];
+        }
+        return subset;
+    }
+
+    /// --------------------------------------------------------------------
+    /// --------------------------------------------------------------------
+    /// ------------------------- Private functions ------------------------
+    /// --------------------------------------------------------------------
+    /// --------------------------------------------------------------------
+
+    /// @notice to be used by the build function to create a governance proposal
```

Comment:
I'm not sure if this language is accurate for us. 

What makes more sense to me would be something like:

```
to be used by the build function to capture the state changes applied by a given task. These state changes will inform whether or not the task will be executed onchain.
```